src/filter/json_encode_filter.rs

```diff
@@ -0,0 +1,133 @@
+/// This filter takes LogLines and encodes them into JSON, emitting the encoded
+/// event as a Raw event. This allows further filters or sinks to operate on the
+/// JSON without needing to understand a LogLine event in particular.
+/// If the LogLine value is a valid JSON object and parse_line config option is true,
+/// then the JSON will be merged with LogLine metadata. Otherwise, the original line
+/// will be included simply as a string.
+pub struct JSONEncodeFilter {
+    parse_line: bool,
+}
+
+/// Configuration for `JSONEncodeFilter`
+#[derive(Clone, Debug)]
+pub struct JSONEncodeFilterConfig {
+    /// The filter's unique name in the routing topology.
+    pub config_path: Option<String>,
+    /// The forwards along which the filter will emit its `metric::Event`s.
+    pub forwards: Vec<String>,
+    /// Whether the filter should attempt to parse LogLine values that are valid JSON objects.
+    pub parse_line: bool,
+}
+
+impl JSONEncodeFilter {
+    /// Create a new JSONEncodeFilter
+    pub fn new(config: &JSONEncodeFilterConfig) -> JSONEncodeFilter {
+        JSONEncodeFilter {
+            parse_line: config.parse_line,
+        }
+    }
+}
+
+impl filter::Filter for JSONEncodeFilter {
+    fn process(
+        &mut self,
+        event: metric::Event,
+        res: &mut Vec<metric::Event>,
+    ) -> Result<(), filter::FilterError> {
+        match event {
+            metric::Event::Telemetry(_) => {
+                // Drop telemetry
+            },
+            metric::Event::Log(l) => if let Some(ref log) = *l {
+                let naive_time = NaiveDateTime::from_timestamp(log.time, 0);
+                let utc_time: DateTime<Utc> = DateTime::from_utc(naive_time, Utc);
+                let metadata = json!({
+                    "time": utc_time.to_rfc3339(),
+                    "path": log.path.clone(),
+                    "tags": log.tags,
+                });
+                let metadata = metadata.as_object().unwrap();
+                // If parse_line is true, and line is parsable as a JSON object, parse it.
+                // Otherwise get an object containing the original line.
+                let value = (if self.parse_line { Some(()) } else { None })
+                    .and_then(|_| serde_json::from_str::<Value>(&log.value).ok())
+                    .and_then(|v| if let Value::Object(obj) = v { Some(obj) } else { None })
+                    .map(|v| { JSON_ENCODE_LOG_PARSED.fetch_add(1, Ordering::Relaxed); v })
+                    .unwrap_or_else(||
+                        if let Value::Object(obj) = json!({"message": log.value.clone()}) {
+                            obj
+                        } else {
+                            unreachable!()
+                        }
+                    );
+                // Combine our various sources of data. Metadata overrides parsed values overrides user-specified.
+                let value = merge_objects(&[&metadata, &log.fields.clone().into(), &value]);
```

Comment:
I think as a user I'd find this confusing. If my json blob had "time" in it then it'd never come through on the other side. This'd be especially tricky if it _mostly_ did roll through the other side. 

What's stopping us from _adding_ to parsed structures? So, make a `__cernan_metadata` or whatever name field in the parsed blob that contained the contents of metadata? The users' data comes through the other side unchanged, we get our data too. 
Maybe

Reply:
The primary use-case of this is for searching in logs. In that context the key names need to be short, simple and memorable.
There's probably a good argument for allowing the user to override cernan's metadata rather than vice versa - eg. they probably know the true timestamp of the event better than we do (our view of time is just when we read it).
In fact, yeah I think that ordering makes more sense. The user knows best, anything we can clean from filters (fields) is second best, and our own metadata is third.

Reply:
> The primary use-case of this is for searching in logs. In that context the key names need to be short, simple and memorable.

That's fair. Once it's merged in master it's not really about the originating use-case anymore and can be applied in surprising ways. That's the perch I'm sitting on when discussing stuff, anyhow. 

> There's probably a good argument for allowing the user to override cernan's metadata rather than vice versa - eg. they probably know the true timestamp of the event better than we do (our view of time is just when we read it).

That's possible to do with a downstream filter. Possible I'm missing some further elaboration you're imagining. 

> In fact, yeah I think that ordering makes more sense. The user knows best, anything we can clean from filters (fields) is second best, and our own metadata is third.

I guess I'm still a little worried about overwriting fields in a payload rather than offsetting them in some way. Adding fields, sure, that's okay but overwriting strikes me as something that could cause surprises, especially for some user that doesn't have logging in mind.